frontends/web/src/components/forms/input-number.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ChangeEvent, useCallback } from 'react';
+import { Input, TInputProps } from './input';
+
+type Props = Omit<TInputProps, 'ref' | 'onInput'> & {
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void
+}
+
+export const NumberInput = (({
+  onChange,
+  ...props
+}: Props) => {
+
+  // accept pasting of various different formats
+  const handlePaste = useCallback((event: React.ClipboardEvent<HTMLInputElement>) => {
+    if (!event.currentTarget || !event.clipboardData) {
+      return;
+    }
+    const text = (
+      event.clipboardData
+        .getData('text')
+        .trim()
+        // remove thousand separator characters space (fr, ru, sw) and apostrophe (ch, li)
+        .replace(/[ ']/g, '')
+    );
+
+    const target = event.currentTarget;
+    const dividedByComma = text.split(',');
+    const dividedByDot = text.split('.');
+    // see if this gets us a valid number
+    const value = Number(text);
+    if (value && !Number.isNaN(value)) {
```

Comment:
sorry I haven't replied to this question...

This condition should set the value if the manipulated text converts to a number. Please note that the text variable is already modified, spaces and apostrophes are stripped off.

In case of `''` or `'0'` the native event should just continue without manipulation.